### PR TITLE
VIC-284-Berater-can-create-Group-chat-available-together-with-regular-chat-

### PR DESCRIPTION
**Pull request by patric-dosch-vi**

### CODE REVIEW

File: src/main/java/de/caritas/cob/userservice/api/facade/CreateChatFacade.java [hunk 1]
```diff
@@ -45,26 +48,49 @@ public class CreateChatFacade {
...
       throw e;
     }
   }
 
+  private String generateChatUrl(Chat chat) {
+    return chat.getConsultingTypeId() != null ? userHelper.generateChatUrl(chat.getId(),
+        chat.getConsultingTypeId()) : StringUtils.EMPTY;
+  }
+
+  private Chat createChatRelation(Consultant consultant, ChatDTO chatDTO) {
```
Comment (tkuzynow) [inline src/main/java/de/caritas/cob/userservice/api/facade/CreateChatFacade.java]: I suggest to rename this method to saveChat

File: src/main/java/de/caritas/cob/userservice/api/facade/CreateChatFacade.java [hunk 2]
```diff
@@ -113,12 +139,29 @@ private String createRocketChatGroup(ChatDTO chatDTO, Chat chat) {
     }
   }
 
+  private Chat convertChatDTOtoChat(ChatDTO chatDTO, Consultant consultant) {
```
Comment (tkuzynow) [inline src/main/java/de/caritas/cob/userservice/api/facade/CreateChatFacade.java]: it is better to create a separate Converter class that can be separately unit tested

File: src/test/java/de/caritas/cob/userservice/api/adapters/web/controller/UserControllerChatE2EIT.java [hunk 3]
```diff
@@ -189,6 +191,42 @@ void reset() {
...
+
+    mockMvc.perform(
+            post("/users/chat/v2/new")
+                .cookie(CSRF_COOKIE)
+                .header(CSRF_HEADER, CSRF_VALUE)
+                .header("rcToken", RandomStringUtils.randomAlphabetic(16))
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(VALID_CREATE_CHAT_BODY)
+                .accept(MediaType.APPLICATION_JSON))
+        .andExpect(status().isCreated());
```
Comment (tkuzynow) [inline src/test/java/de/caritas/cob/userservice/api/adapters/web/controller/UserControllerChatE2EIT.java]: could you please add some assertions with jsonPath to the response? For the most important attributes

File: src/test/java/de/caritas/cob/userservice/api/facade/CreateChatV2FacadeTest.java [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+import org.junit.runner.RunWith;
+import org.mockito.InOrder;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.slf4j.Logger;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+public class CreateChatV2FacadeTest {
```
Comment (tkuzynow) [inline src/test/java/de/caritas/cob/userservice/api/facade/CreateChatV2FacadeTest.java:50]: we should use unit5 framework for tests of the newly implemented classes

File: src/main/java/de/caritas/cob/userservice/api/facade/CreateChatFacade.java (hunk 1 above)
Comment (patric-dosch-vi) [reply src/main/java/de/caritas/cob/userservice/api/facade/CreateChatFacade.java]: thanks, I wanted, but at fist I needed a extra name ;)

**Comment by tkuzynow** [comment]
I like the code, is well structured and clean, easily readible. I've added just a couple of small comments, please have a look

File: src/main/java/de/caritas/cob/userservice/api/facade/CreateChatFacade.java (hunk 1 above)
Comment (patric-dosch-vi) [reply src/main/java/de/caritas/cob/userservice/api/facade/CreateChatFacade.java]: now I remember... I have then createChat as the generic method that takes the function. And createChat that is the function. I was also unhappy with that.

File: src/test/java/de/caritas/cob/userservice/api/facade/CreateChatV2FacadeTest.java (hunk 4 above)
Comment (patric-dosch-vi) [reply src/test/java/de/caritas/cob/userservice/api/facade/CreateChatV2FacadeTest.java:50]: it's a copy of the CreateChatV1FacadeTest what was already there. As I would re-write a lot of stuff there, I would not agree as I would normally.

Comment (tkuzynow) [reply src/test/java/de/caritas/cob/userservice/api/facade/CreateChatV2FacadeTest.java:50]: ok

File: src/main/java/de/caritas/cob/userservice/api/facade/CreateChatFacade.java (hunk 1 above)
Comment (patric-dosch-vi) [reply src/main/java/de/caritas/cob/userservice/api/facade/CreateChatFacade.java]: for me its fine with your change.

File: src/main/java/de/caritas/cob/userservice/api/facade/CreateChatFacade.java (hunk 2 above)
Comment (patric-dosch-vi) [reply src/main/java/de/caritas/cob/userservice/api/facade/CreateChatFacade.java]: done

File: src/test/java/de/caritas/cob/userservice/api/adapters/web/controller/UserControllerChatE2EIT.java (hunk 3 above)
Comment (patric-dosch-vi) [reply src/test/java/de/caritas/cob/userservice/api/adapters/web/controller/UserControllerChatE2EIT.java]: done. will change maybe later, when its more clear what about the Chat-Link

**Review by tkuzynow — DISMISSED** [review]

**Review by tkuzynow — APPROVED** [review]

**Pull request merged**